File: 0002-add-two-numbers/0002-add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* dummy = new ListNode(-1);
        ListNode * cpy = dummy;
        ListNode * temp1= l1;
        ListNode* temp2=l2;
        int caary=0;
        while(temp1!=NULL||temp2!=NULL){
            int sum=caary;
            if(temp1) {sum +=temp1->val;}
            if(temp2){sum +=temp2->val;}
            ListNode* newNode= new ListNode(sum%10);
            caary=sum/10;

            cpy->next=newNode;
            cpy=cpy->next;

            if(temp1) {temp1=temp1->next;}
            if(temp2) {temp2=temp2->next;}
        }
        if(caary){
            ListNode *newNode = new ListNode(caary);
            cpy->next=newNode;
        } 
        return dummy->next;
             
    }
};

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode()
        tail = dummy
        chk1 = l1
        chk2 = l2 
        carry = 0
        while(chk1 or chk2):
            sumi = carry
            if chk1 :
                sumi+=chk1.val
            if chk2:
                sumi +=chk2.val
            tail.next = ListNode(sumi%10)
            carry = sumi//10
            tail = tail.next
            if chk1 :
                chk1=chk1.next
            if chk2: 
                chk2=chk2.next

        if carry:
            tail.next = ListNode(carry)
        return dummy.next
